### ISSUE TITLE
请教下GRPO训练时出现模型多次异常触碰到Max_length的问题

### ISSUE THREAD

**Issue by 1212wuhu** [modelscope/ms-swift#4758]
在我进行grpo训练时，模型会多次出现过长的生成结果，但这种情况一般是周期性的，直至训练至1200步之后，情况好像不太受控制了：

![Image](https://github.com/user-attachments/assets/12e0d731-7643-4443-999e-724478408737)
附上loss和grad_norm变化图

![Image](https://github.com/user-attachments/assets/f32c40e0-2adb-44b9-b639-e4ea91455912)

可以看到，max_length异常升高之后无法再有效降低，FormatCorrectnessReward和AnswerAccuracyReward都出现下降，直至最后也很难恢复正常。请问这种情况是否要从上一个检查点重新训练？有什么方式能够抑制这种情况出现吗？

下面是我的脚本
```
# 4*80G GPU

CUDA_VISIBLE_DEVICES=1,2,3 \
NPROC_PER_NODE=3 \
swift rlhf \
    --rlhf_type grpo \
    --model /root/autodl-tmp/landai_0610_sft \
    --train_type full \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --dataset /root/autodl-tmp/mixed_dapo_train_24376s_add_solution.jsonl \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --max_length 12000 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 4 \
    --gradient_accumulation_steps 4 \
    --eval_steps 600 \
    --save_steps 200 \
    --learning_rate 1e-6 \
    --save_total_limit 1 \
    --logging_steps 1 \
    --output_dir output_grpo \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --max_completion_length 8000 \
    --external_plugins /root/autodl-tmp/ms-swift/examples/train/grpo/plugin/plugin.py \
    --reward_funcs custom_format_correctness custom_answer_accuracy \
    --reward_weights 1 1 \
    --num_generations 12 \
    --system /root/autodl-tmp/ms-swift/examples/train/grpo/prompt.txt \
    --deepspeed zero3 \
    --temperature 1.0 \
    --repetition_penalty 1.1 \
    --async_generate false \
    --padding_free true \
    --report_to swanlab \
    --log_completions true \
    --overlong_filter true \
    --num_iterations 1 \
    --max_steps 15000 \
    --attn_impl flash_attn 
```

**Comment by 1212wuhu** [comment]
我从1000的检查点重新启动了训练，可是结果好像是一样的，还是出现较大的奖励下降和模型回复多次触碰到最大限度值的情况：

![Image](https://github.com/user-attachments/assets/bf4f94e9-1a04-4468-ac4c-a7b2e5b208c6)

我应该怎么解决呢

**Comment by hjh0119** [comment]
maybe a length related reward can help

**Comment by 1212wuhu** [comment]
![Image](https://github.com/user-attachments/assets/afab513d-a798-4181-9b1d-6910706c8f1b)

![Image](https://github.com/user-attachments/assets/d8d93af3-8987-4cb9-b6dd-aa50fc3c0852)

![Image](https://github.com/user-attachments/assets/cf573fbc-973a-4cee-9410-c3ee22019082)

@hjh0119 我重启了训练，并使用了dapo算法，可是看着并没有明显改善，新脚本如下：
```
# 4*80G GPU

CUDA_VISIBLE_DEVICES=1,2 \
NPROC_PER_NODE=2 \
swift rlhf \
    --rlhf_type grpo \
    --model /root/autodl-tmp/1111 \
    --train_type full \
    --use_vllm true \
    --vllm_mode server \
    --vllm_server_host 127.0.0.1 \
    --vllm_server_port 8000 \
    --dataset /root/autodl-tmp/mixed_dapo_train_14593s_1_1_add_solution.jsonl \
    --torch_dtype bfloat16 \
    --num_train_epochs 1 \
    --max_length 12000 \
    --per_device_train_batch_size 1 \
    --per_device_eval_batch_size 6 \
    --gradient_accumulation_steps 6 \
    --eval_steps 1000 \
    --save_steps 500 \
    --learning_rate 1e-6 \
    --save_total_limit 2 \
    --logging_steps 1 \
    --output_dir output_grpo \
    --warmup_ratio 0.01 \
    --dataloader_num_workers 4 \
    --max_completion_length 8000 \
    --loss_type bnpo \
    --epsilon_high 0.28 \
    --dynamic_sample true \
    --max_resample_times 3 \
    --soft_cache_length 4096 \
    --external_plugins /root/autodl-tmp/ms-swift/examples/train/grpo/plugin/plugin.py \
    --reward_funcs custom_format_correctness custom_answer_accuracy soft_overlong \
    --reward_weights 1 1 1 \
    --num_generations 12 \
    --system /root/autodl-tmp/ms-swift/examples/train/grpo/prompt.txt \
    --deepspeed zero3 \
    --temperature 1.0 \
    --repetition_penalty 1.1 \
    --async_generate false \
    --padding_free true \
    --report_to swanlab \
    --log_completions true \
    --overlong_filter true \
    --max_grad_norm 0.5 \
    --attn_impl flash_attn 
```
请问我是否需要放弃之前强化学习的结果，从0开始呢？还是说静静等待模型重新涨reward？
